packages/fuels-test-helpers/src/lib.rs

```diff
@@ -188,11 +188,10 @@ pub async fn setup_test_client(
     let config_with_coins: Value =
         serde_json::from_str(result.as_str()).expect("Failed to build config_with_coins JSON");
 
-    let srv_address = if let Some(node_config) = node_config {
-        node_config.addr
-    } else {
-        let free_port = pick_unused_port().expect("No ports free");
-        SocketAddr::new("127.0.0.1".parse().unwrap(), free_port)
+    let srv_address = match node_config {
+        Some(config) if config.addr.port() != 0 && is_free(config.addr.port()) => config.addr,
+        Some(config) if !is_free(config.addr.port()) => panic!("Error: Address already in use"),
```

Comment:
not sure why would this be useful since this would panic fail anyway if the port was already taken

Reply:
https://github.com/FuelLabs/fuels-rs/pull/425#discussion_r906842007